src/util/util_shared_res.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <cstdint>
+
+#include "./com/com_include.h"
+
+#include "../d3d9/d3d9_common_texture_desc.h"
+#include "../d3d11/d3d11_common_texture_desc.h"
+
+namespace dxvk::sharedRes {
+
+    HANDLE openKmtHandle(HANDLE kmt_handle);
+
+    bool setMetadata(HANDLE handle, void *buf, uint32_t bufSize);
+    bool getMetadata(HANDLE handle, void *buf, uint32_t bufSize, uint32_t *metadataSize);
+
+    struct DxvkSharedMetadata {
+      enum class DxvkSharedMetadataSourceApi{
+        D3D9,
+        D3D11,
+      } sourceApi;
+
+      union {
+        D3D9_COMMON_TEXTURE_DESC d3d9;
+        D3D11_COMMON_TEXTURE_DESC d3d11;
```

Comment:
Can we just make this a new struct instead and translate to and from it in the D3D11 and D3D9 frontends, respectively, so that we won't end up with a combinatorial explosion of metadata translations? D3D12 could share that format.

This could be pretty much a copy of `D3D11_COMMON_TEXTURE_DESC`, except without the `Depth` field since these APIs do not allow sharing 3D textures. If we ever have to implement D3D11<->Vulkan/D3D12 sharing, we'll have to make significant changes to the entire thing anyway, so making this a union is not immediately beneficial.